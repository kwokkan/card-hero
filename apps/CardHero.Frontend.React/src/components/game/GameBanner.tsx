import React from "react";

interface IGameBannerProps {
    visible: boolean;
    message: string;
    additionalClasses?: string;
}

export function GameBanner(props: IGameBannerProps) {
    if (!props.visible) {
        return null;
    }

    let messageClass = "game-banner-message";

    if (props.additionalClasses) {
        messageClass += " " + props.additionalClasses;
    }

    return (
        <div className="game-banner">
            <span className={messageClass}>
                {props.message}
            </span>
        </div>
    );
}
