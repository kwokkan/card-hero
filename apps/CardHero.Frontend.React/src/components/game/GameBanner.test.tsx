import renderer from "react-test-renderer";
import { GameBanner } from "./GameBanner";

test("<GameBanner /> renders null when not visible", () => {
    const tree = renderer
        .create(
            <GameBanner visible={false} message="My message" />
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("<GameBanner /> renders with message", () => {
    const tree = renderer
        .create(
            <GameBanner visible={true} message="My message" />
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("<GameBanner /> renders with message and additional class", () => {
    const tree = renderer
        .create(
            <GameBanner visible={true} message="My message" additionalClasses="new-class more-class" />
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
