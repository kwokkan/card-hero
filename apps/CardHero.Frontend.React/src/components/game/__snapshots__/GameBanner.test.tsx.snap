// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameBanner /> renders null when not visible 1`] = `null`;

exports[`<GameBanner /> renders with message 1`] = `
<div
  className="game-banner"
>
  <span
    className="game-banner-message"
  >
    My message
  </span>
</div>
`;

exports[`<GameBanner /> renders with message and additional class 1`] = `
<div
  className="game-banner"
>
  <span
    className="game-banner-message new-class more-class"
  >
    My message
  </span>
</div>
`;
